crates/uv-shell/src/windows.rs

```diff
@@ -24,34 +23,27 @@ pub fn prepend_path(path: &Path) -> anyhow::Result<bool> {
     let windows_path = get_windows_path_var()?;
 
     // Add the new path to the existing `PATH` variable.
-    let windows_path = windows_path.and_then(|windows_path| {
-        prepend_to_path(windows_path, OsString::from(path).encode_wide().collect())
-    });
-
+    let windows_path =
+        windows_path.and_then(|windows_path| prepend_to_path(&windows_path, HSTRING::from(path)));
     // If the path didn't change, then we don't need to do anything.
     let Some(windows_path) = windows_path else {
         return Ok(false);
     };
 
     // Set the `PATH` variable in the registry.
-    apply_windows_path_var(windows_path)?;
+    apply_windows_path_var(&windows_path)?;
 
     Ok(true)
 }
 
 /// Set the windows `PATH` variable in the registry.
-fn apply_windows_path_var(path: Vec<u16>) -> anyhow::Result<()> {
-    let root = RegKey::predef(HKEY_CURRENT_USER);
-    let environment = root.open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)?;
+fn apply_windows_path_var(path: &HSTRING) -> anyhow::Result<()> {
+    let environment = CURRENT_USER.create("Environment")?;
```

Comment:
Not sure why this `create` and not `open`, I've taken this from rustup

Reply:
This might actually be https://github.com/rust-lang/rustup/issues/3779, so this PR also fixes a bug.